data_processing/cleaning_demographic_data.Rmd

```diff
@@ -52,11 +50,412 @@ demo_dat %>%
+                                      str_starts(race_2, "asian") ~ "asian/asian american",
+                                      str_starts(race_2, "bla") ~ "black/african american/african",
+                                      str_starts(race_2, "pacific i") ~ "native hawaiian/pacific islander",
+                                      str_starts(race_2, "wh") ~ "white",
+                                   str_detect(race_2, "hisp") ~ "unknown",
+                                   str_detect(race_2, "unknown|unable|refused|not collected|applicable|verified|#|client|mex") ~ "unknown",
+                                   is.na(race_2) ~ "unknown",
+                                      
+                                         TRUE ~ race_2))
+```
+
+check for race_2
+```{r}
+demo_dat %>% 
+  count(race_2, race_2_recode)
+```
+
+
+Ethnicity - replace all instances of different spellings or orders of hispanic/latino/a/x to hispanic/latinx. Changing mexican/chicano, cuban, other hispanic/latino, and puerto rican to hispanic/latinx. Also, replace all instances of different spellings or orders of not hispanic/latino to non-hispanic/latinx 
+
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(ethnicity_recode = case_when(str_starts(ethnicity, "no") ~ "non-hispanic/latinx",
+                                      str_detect(ethnicity,"hispa") ~ "hispanic/latinx", 
+                                      str_detect(race_2, "not hispanic|no hispanic") ~ "non-hispanic/latinx",
+                                      str_detect(race_2, "hispanic") ~ "hispanic/latinx",
+                                      str_detect(race_2, "mex") ~ "hispanic/latinx",
+                                      str_detect(ethnicity, "mex|puerto|cuban") ~ "hispanic/latinx",
+                                      str_detect(ethnicity, "unknown|#|applicable|data not|client|missing|rent") ~ "unknown",
+                                      is.na(ethnicity) ~ "unknown",
+                                      
+                                      
+                                         TRUE ~ ethnicity))
+```
+
+check for ethnicity 
+```{r}
+demo_dat %>% 
+  count(ethnicity, race_2, ethnicity_recode) 
+```
+
+current_martial_status - replace unknowns and data not collected with "unknown". 
+
+### Question: How would we like to recode domestic partnership? (Lump with married?) What to do with single? (100/103 records with single are from LA) ALSO, what to do with "no"? (both from Merced, might get fixed when column shift issue is solved)
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(current_marital_recode = case_when(str_detect(current_marital_status, "unknown|#|applicable|data not") ~ "unknown",
+                                            str_detect(current_marital_status, "domestic|single") ~ "flag",
+                                      is.na(current_marital_status) ~ "unknown",
+                                         TRUE ~ current_marital_status))
+```
+
```

Comment:
Good question--let's table this for after the data dump. I think we can collapse domestic partnership w/ married if the counts are very low.